src/VASSAL/build/module/map/BoardPicker.java

```diff
@@ -413,6 +413,17 @@ public void actionPerformed(ActionEvent e) {
     }
   }
 
+  /**
+   * @return an array of the names of all boards from which have been selected either by the user via the dialog or from reading a savefile
+   */
+  public String[] getSelectedBoardNames() {
```

Comment:
it's common to use java.util.List in Java:
`public List<String> getSelectedBoardNames { List<String> result = new ArrayList<>(); /* ... */ return result; }`

or using the new stream API:
`public List<String> getSelectedBoardNames { boards.stream().map(Board::getName).collect(Collectors.toList()); }`

The caller can then do:
`List<String> boardNames = map.getBoardPicker().getSelectedBoardNames(); if (!boardNames.isEmpty()) owningBoardName = boardNames.get(0);`

Reply:
Ok, thanks, will do.
I was just trying to retain the code style the same as the existing getAllowableBoardNames and getAllowableLocalizedBoardNames classes in this file

Reply:
I am not one of the maintainers and don't really have a say in these matters but I would not add more bad code to already existing bad code, especially if it is a public method that will potentially become public API which module developers will use in their modules.

Reply:
Yes, for new methods, use Lists rather than arrays.  If you want to go further and refactor existing methods, just remember that an unknown amount of existing custom code in modules may use any non-private methods, so @Deprecate the existing method that returns an array and create a new method that returns a List.

Reply:
Roger Wilco.
I think if refactoring further methods, I'll do so in a separate PR.
I've noticed I've managed to create this PR against my master branch, so I'm probably going to close this PR this evening, and recreate it against a proper local branch... (unless it gets merged before so)